contracts/lib/NonceLib.sol

```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.26;
+
+import { MODE_MODULE_ENABLE } from "contracts/types/Constants.sol";
+
+library NonceLib {
+    function getValidator(uint256 nonce) internal pure returns (address validator) {
+        assembly {
+            validator := shr(96, shl(32, nonce))
```

Comment:
would be great to add some inline comment // to briefly explain what we're trying to achieve there

Reply:
added the nonce struct => the assembly code is easier to understand